Fix grammar and construction in 2.1.10 (#1672)
javascript-tutorial/en.javascript.info: 2-ui/1-document/10-size-and-scroll-window/article.md

How do we find the width and height of the browser window? How do we get the full width and height of the document, including the scrolled out part? How do we scroll the page using JavaScript?

> For these, we can use the root document element `document.documentElement`, which corresponds to the `<html>` tag. But there are additional methods and peculiarities to consider.

Reviewer: I would leave this as is, or maybe replace "most such requests," with "for this type of information" (probably a little better w/out comma). And I prefer "that" over "which".

Reviewer: Breaking the "that" clause out into a separate sentence doesn't seem as good to me as way it is in the original. Still favor my original suggestion.

Author: ```suggestion
For this type of information, we can use the root document element `document.documentElement`, that corresponds to the `<html>` tag. But there are additional methods and peculiarities to consider.
```

<button onclick="alert(document.documentElement.clientHeight)">alert(document.documentElement.clientHeight)</button>
```

````warn header="Not `window.innerWidth/innerHeight`"
Browsers also support properties like `window.innerWidth/innerHeight`. They look like what we want, so why not to use them instead?

If there exists a scrollbar, and it occupies some space, `clientWidth/clientHeight` provide the width/height without it (subtract it). In other words, they return the width/height of the visible part of the document, available for the content.

`window.innerWidth/innerHeight` includes the scrollbar.

> If there's a scrollbar, and it occupies some space, then these two lines return different values:

Reviewer: I would stick with "show", especially since "return" has special meaning.

Author: Good point. Will bring it back.

Author: ```suggestion
If there's a scrollbar, and it occupies some space, then these two lines show different values:
```

```js run
alert( window.innerWidth ); // full window width
alert( document.documentElement.clientWidth ); // window width minus the scrollbar
```

> In most cases, we need the *available* window width—to draw or position something within scrollbars if there are any. So we should use `documentElement.clientHeight/clientWidth`.

Reviewer: Suggest: "In most cases we need the *available* window width, in order to draw or position something within the scrollbars (if there are any). So we should use `documentElement.clientHeight/clientWidth`."

Reviewer: Could go with or without the comma there. No strong preference.

````

```warn header="`DOCTYPE` is important"

> Please note: top-level geometry properties may work a little bit differently when there's no `<!DOCTYPE HTML>` in HTML.

Reviewer: I like the chatty little addition. Would leave it in.

Reviewer: Doesn't bother me the way it was, and the statement that "JavaScript can be quite finicky" seems to go beyond the scope of what is being discussed.

Author: ```suggestion
Please note: top-level geometry properties may work a little bit differently when there's no `<!DOCTYPE HTML>` in HTML. Odd things are possible.
```

alert('Full document height, with scrolled out part: ' + scrollHeight);
```

> Why so? Well, the gist of it is these inconsistencies come from ancient times.

Reviewer: Again, would leave as is. Humanizing. Even if 'not a "smart" logic' is a bit unusual, grammatically.

Author: It's too awkward I think. I found it to be inconsistent with the rest of the writing style in the article, especially when read top-down. Maybe we can rephrase it to be more idiomatic?

```suggestion
Why, you might ask? These inconsistencies come from ancient times, not bad design.
```

Reviewer: Either way. I still prefer the original, though.

Author: ```suggestion
Why so? Better don't ask. These inconsistencies come from ancient times, not a "smart" logic.
```
